pkg/network/protocols/redis/model_linux.go

```diff
@@ -0,0 +1,111 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package redis
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/network/protocols"
+	"github.com/DataDog/datadog-agent/pkg/network/types"
+)
+
+// EventWrapper wraps an ebpf event and provides additional methods to extract information from it.
+// We use this wrapper to avoid recomputing the same values (key name) multiple times.
+type EventWrapper struct {
+	*EbpfEvent
+
+	keyNameSet bool
+	keyName    string
+	commandSet bool
+	command    CommandType
+}
+
+// NewEventWrapper creates a new EventWrapper from an ebpf event.
+func NewEventWrapper(e *EbpfEvent) *EventWrapper {
+	return &EventWrapper{EbpfEvent: e}
+}
+
+// ConnTuple returns the connection tuple for the transaction
+func (e *EventWrapper) ConnTuple() types.ConnectionKey {
+	return types.ConnectionKey{
+		SrcIPHigh: e.Tuple.Saddr_h,
+		SrcIPLow:  e.Tuple.Saddr_l,
+		DstIPHigh: e.Tuple.Daddr_h,
+		DstIPLow:  e.Tuple.Daddr_l,
+		SrcPort:   e.Tuple.Sport,
+		DstPort:   e.Tuple.Dport,
+	}
+}
+
+// getFragment returns the actual query fragment from the event.
+func getFragment(e *EbpfTx) []byte {
+	if e.Buf_len == 0 {
+		return nil
+	}
+	if e.Buf_len > uint16(len(e.Buf)) {
+		return e.Buf[:len(e.Buf)]
+	}
+	return e.Buf[:e.Buf_len]
+}
+
+// KeyName returns the key name of the key.
+func (e *EventWrapper) KeyName() string {
+	if !e.keyNameSet {
+		e.keyName = string(getFragment(&e.Tx))
```

Comment:
nit: can be done a future PR
we should be using string interning here to spare redundant memory